### ISSUE TITLE
Feature Request: Default Repo Setting

### ISSUE THREAD

**Issue by irrlphnt** [kylecrawshaw/ImagrAdmin#14]
- [x] I have searched [existing issues](https://github.com/kylecrawshaw/ImagrAdmin/issues?q=is%3Aissue+sort%3Acreated-desc) and **this is not a duplicate**.
## Question or Feature Request

Would it be possible to set a default repo location (or locations...) in the app? Currently having to type out (or paste) the full path to add scripts seems like an opportunity for typos or mistakes...

Thanks for your great work so far!

Peter

**Comment by kylecrawshaw** [comment]
@irrlphnt -- I was thinking about this during some testing and I think a feature like this could certainly help remove room for error, but implementation will be slightly tricky to do.I'll play around to see how I can present this in the UI without causing any confusion for the user. It could be something set in a preferences window, but there currently are no other preferences so it may be a bit odd having only a single preference that can be set. Then there is the question of how is this presented in any of the component UI's without adding additional form fields. If you have any ideas of what you would expect as a user please feel free to share them here.

**Comment by irrlphnt** [comment]
@kylecrawshaw You're right: it doesn't seem right for app-wide preferences. Thinking about it just now, I wonder whether a custom key within the imagr_config.plist might be a good way of doing it, as well as making it usable by other interested parties. Maybe it's actually a suggestion for Graham...

**Comment by erikng** [comment]
While that sounds good (adding it to Imagr) people will sort their files in whatever manner they want. For example I have a "dmg" "pkg" and "scripts" folder and would want to expand each path. I'm not sure how you could account for it except to possibly write a value for each possible type and optionally have the admin set them.

Sent from my iPhone

On Aug 1, 2016, at 6:29 AM, Peter Wilcox <notifications@github.com<mailto:notifications@github.com>> wrote:

@kylecrawshawhttps://github.com/kylecrawshaw You're right: it doesn't seem right for app-wide preferences. Thinking about it just now, I wonder whether a custom key within the imagr_config.plist might be a good way of doing it, as well as making it usable by other interested parties. Maybe it's actually a suggestion for Graham...

##

**Comment by kylecrawshaw** [comment]
@irrlphnt and @erikng, I've been thinking about this a bit. I could possibly implement some sort of component "library", so you can add all of your scripts and packages and then when you're adding a new component to a workflow you could either pull from the library or create a new one with blank values. There could be support in the "library" for components, urls to scripts/paths and actual scripts you could add. This doesn't exactly address what you were looking for, but I think it could be one way to limit some repetition. Thoughts? Does that idea make sense?

**Comment by erikng** [comment]
Sounds interesting, but then how do you update the library? I don't like the idea of manually adding/managing files into anything.

Sent from my iPad

On Aug 2, 2016, at 8:58 PM, kylecrawshaw <notifications@github.com<mailto:notifications@github.com>> wrote:

@irrlphnthttps://github.com/irrlphnt and @eriknghttps://github.com/erikng, I've been thinking about this a bit. I could possibly implement some sort of component "library", so you can add all of your scripts and packages and then when you're adding a new component to a workflow you could either pull from the library or create a new one with blank values. There could be support in the "library" for components, urls to scripts/paths and actual scripts you could add. This doesn't exactly address what you were looking for, but I think it could be one way to limit some repetition. Thoughts? Does that idea make sense?

##

**Comment by kylecrawshaw** [comment]
Completely agreed. It would be tedious to manually update everything, but there are too many edges cases and you would still need some sort of metadata file.

**Comment by erikng** [comment]
Perhaps the best option here is to specify one path and iterate through to find each file type. If there isn't any file type, attempt to read the first line and process the type (i.e. Scripts).

It wouldn't account for the super edge case of someone hosting scripts on some arbitrary path, but I'm not really sure how to solve it. Perhaps this needs more thought before diving in.

On Aug 2, 2016, at 9:05 PM, kylecrawshaw <notifications@github.com<mailto:notifications@github.com>> wrote:

Completely agreed. It would be tedious to manually update everything, but there are too many edges cases and you would still need some sort of metadata file.

##
